test/bucket.ts

```diff
@@ -281,7 +281,9 @@ describe('Bucket', () => {
       });
 
       const bucket = new Bucket(storageInstance, BUCKET_NAME);
-      assert(bucket instanceof ServiceObject);
+      // Using assert.strictEqual instead of assert to prevent
+      // coercing of types.
+      assert.strictEqual(bucket instanceof ServiceObject, true);
```

Comment:
This causes the compiler to recognise `bucket` as type `ServiceObject`, rather than `any`, which causes the line below to fail.

I'm starting to think we should probably rewrite our tests in `sinon`.